lib/getNextJobs.js

```diff
@@ -1,6 +1,28 @@
 'use strict';
 
 const { PR_JOB_NAME, EXTERNAL_TRIGGER_ALL } = require('screwdriver-data-schema').config.regex;
+const STAGE_SETUP_PATTERN = /^stage@([\w-]+)(?::setup)$/;
+
+/**
+ * Check if the job is setup job with setup suffix
+ * @param  {String} jobName                 Job name
+ * @return {Boolean}
+ */
+function isStageSetup(jobName) {
+    return STAGE_SETUP_PATTERN.test(jobName);
+}
+
+/**
+ * Check if the job is a stage job
+ * @param  {String} jobName                 Job name
+ * @param  {Object} workflowGraph           Workflow Graph
+ * @return {Boolean}
+ */
+function isStageJob(workflowGraph, jobName) {
+    const jobNode = workflowGraph.nodes.find(n => n.name === jobName);
```

Comment:
We may not find a match in some scenarios. Ex: when event `startFrom` is a stage (`stage@production`)
Refer: https://github.com/screwdriver-cd/models/pull/623

We need to return `false` when there is no match found